### ISSUE TITLE
Unable to find PyVmomi installed with pip

### ISSUE THREAD

**Issue by ChrisD1990** [ansible/ansible#44245]
##### SUMMARY

when i find vmware state with ansible, it will told me .

```
[localhost]: FAILED! => {"changed": false, "msg": "PyVmomi Python module required. Install using \"pip install PyVmomi\""}
```
i have installed PyVmomi  with pip.

```
[root@Georgeoelv1 ansible]# pip install PyVmomi
Requirement already satisfied: PyVmomi in /usr/lib64/python2.7/site-packages (6.7.0)
Requirement already satisfied: six>=1.7.3 in /usr/lib/python2.7/site-packages (from PyVmomi) (1.9.0)
Requirement already satisfied: requests>=2.3.0 in /usr/lib/python2.7/site-packages (from PyVmomi) (2.19.1)
Requirement already satisfied: idna<2.8,>=2.5 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (2.7)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (3.0.4)
Requirement already satisfied: urllib3<1.24,>=1.21.1 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (1.23)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (2018.4.16)
```
below is my yml file :

```yaml
-   - name: Revert the given snapshot for virtual machines
-     hosts: localhost
-     tasks:
-       - name: get vm state
-         vmware_guest_find:
-           hostname: "{{ vcenter_hostname }}"
-           username: "{{ vsphere_user }}"
-           password: "{{ vsphere_pwd }}"
-           name: "{{ vm_name }}"
-           validate_certs: False
-         register: vm_state
-         until: vm_state.rc == 0
-         retries: 10
-         delay: 15
```
##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
ansible 2.6.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Feb 20 2018, 09:19:12) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]**
```

##### OS / ENVIRONMENT

**ANSIBLE_SSH_ARGS(/etc/ansible/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=30m
DEFAULT_EXECUTABLE(/etc/ansible/ansible.cfg) = /bin/sh
DEFAULT_HOST_LIST(/etc/ansible/ansible.cfg) = [u'/etc/ansible/hosts']
DEFAULT_REMOTE_PORT(/etc/ansible/ansible.cfg) = 22
DEFAULT_REMOTE_USER(/etc/ansible/ansible.cfg) = root
DEFAULT_SUDO_USER(/etc/ansible/ansible.cfg) = root
DEFAULT_TIMEOUT(/etc/ansible/ansible.cfg) = 120
DEPRECATION_WARNINGS(/etc/ansible/ansible.cfg) = False**

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```

```

**Comment by ChrisD1990** [comment]
there is error log with -vvvv
TASK [get vm state] *************************************************************************************************************************************************************************************************
task path: /etc/ansible/Vm_rollback_snapshost.yml:5
Using module file /usr/lib/python2.7/site-packages/ansible/modules/cloud/vmware/vmware_guest_find.py
<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 localhost '/bin/sh -c '"'"'echo ~root && sleep 0'"'"''
<localhost> (0, '/root\n', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 localhost '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583 `" && echo ansible-tmp-1534398031.45-118622861370583="` echo /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583 `" ) && sleep 0'"'"''
<localhost> (0, 'ansible-tmp-1534398031.45-118622861370583=/root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583\n', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<localhost> PUT /root/.ansible/tmp/ansible-local-2793f4YTy7/tmpfbtgFz TO /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py
<localhost> SSH: EXEC sftp -b - -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 '[localhost]'
<localhost> (0, 'sftp> put /root/.ansible/tmp/ansible-local-2793f4YTy7/tmpfbtgFz /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py\n', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug2: Remote version: 3\r\ndebug2: Server supports extension "posix-rename@openssh.com" revision 1\r\ndebug2: Server supports extension "statvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "fstatvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "hardlink@openssh.com" revision 1\r\ndebug2: Server supports extension "fsync@openssh.com" revision 1\r\ndebug3: Sent message fd 5 T:16 I:1\r\ndebug3: SSH_FXP_REALPATH . -> /root size 0\r\ndebug3: Looking up /root/.ansible/tmp/ansible-local-2793f4YTy7/tmpfbtgFz\r\ndebug3: Sent message fd 5 T:17 I:2\r\ndebug3: Received stat reply T:101 I:2\r\ndebug1: Couldn\'t stat remote file: No such file or directory\r\ndebug3: Sent message SSH2_FXP_OPEN I:3 P:/root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py\r\ndebug3: Sent message SSH2_FXP_WRITE I:4 O:0 S:32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 4 32768 bytes at 0\r\ndebug3: Sent message SSH2_FXP_WRITE I:5 O:32768 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:6 O:65536 S:31131\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 5 32768 bytes at 32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 6 31131 bytes at 65536\r\ndebug3: Sent message SSH2_FXP_CLOSE I:4\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 localhost '/bin/sh -c '"'"'chmod u+x /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/ /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py && sleep 0'"'"''
<localhost> (0, '', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')

<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 -tt localhost '/bin/sh -c '"'"'/usr/bin/python /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py && sleep 0'"'"''

<localhost> (1, '\r\n{"msg": "PyVmomi Python module required. Install using \\"pip install PyVmomi\\"", "failed": true, "invocation": {"module_args": {"username": "danhua.wang@hpe.com", "datacenter": null, "name": "shcOEL75vm2", "hostname": "selvc01.hpeswlab.net", "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", "validate_certs": false, "port": 443, "uuid": null}}}\r\n', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 1\r\nShared connection to localhost closed.\r\n')
<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 localhost '/bin/sh -c '"'"'rm -f -r /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/ > /dev/null 2>&1 && sleep 0'"'"''
<localhost> (0, '', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "datacenter": null,
            "hostname": "selvc01.hpeswlab.net",
            "name": "shcOEL75vm2",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 443,
            "username": "danhua.wang@hpe.com",
            "uuid": null,
            "validate_certs": false
        }
    },
    "msg": "PyVmomi Python module required. Install using \"pip install PyVmomi\""
}
        to retry, use: --limit @/etc/ansible/Vm_rollback_snapshost.retry

PLAY RECAP **********************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1

**Comment by ChrisD1990** [comment]
```
TASK [get vm state] *************************************************************************************************************************************************************************************************
task path: /etc/ansible/Vm_rollback_snapshost.yml:5
Using module file /usr/lib/python2.7/site-packages/ansible/modules/cloud/vmware/vmware_guest_find.py
<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 localhost '/bin/sh -c '"'"'echo ~root && sleep 0'"'"''
<localhost> (0, '/root\n', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 localhost '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583 `" && echo ansible-tmp-1534398031.45-118622861370583="` echo /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583 `" ) && sleep 0'"'"''
<localhost> (0, 'ansible-tmp-1534398031.45-118622861370583=/root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583\n', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<localhost> PUT /root/.ansible/tmp/ansible-local-2793f4YTy7/tmpfbtgFz TO /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py
<localhost> SSH: EXEC sftp -b - -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 '[localhost]'
<localhost> (0, 'sftp> put /root/.ansible/tmp/ansible-local-2793f4YTy7/tmpfbtgFz /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py\n', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug2: Remote version: 3\r\ndebug2: Server supports extension "posix-rename@openssh.com" revision 1\r\ndebug2: Server supports extension "statvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "fstatvfs@openssh.com" revision 2\r\ndebug2: Server supports extension "hardlink@openssh.com" revision 1\r\ndebug2: Server supports extension "fsync@openssh.com" revision 1\r\ndebug3: Sent message fd 5 T:16 I:1\r\ndebug3: SSH_FXP_REALPATH . -> /root size 0\r\ndebug3: Looking up /root/.ansible/tmp/ansible-local-2793f4YTy7/tmpfbtgFz\r\ndebug3: Sent message fd 5 T:17 I:2\r\ndebug3: Received stat reply T:101 I:2\r\ndebug1: Couldn\'t stat remote file: No such file or directory\r\ndebug3: Sent message SSH2_FXP_OPEN I:3 P:/root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py\r\ndebug3: Sent message SSH2_FXP_WRITE I:4 O:0 S:32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 4 32768 bytes at 0\r\ndebug3: Sent message SSH2_FXP_WRITE I:5 O:32768 S:32768\r\ndebug3: Sent message SSH2_FXP_WRITE I:6 O:65536 S:31131\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 5 32768 bytes at 32768\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: In write loop, ack for 6 31131 bytes at 65536\r\ndebug3: Sent message SSH2_FXP_CLOSE I:4\r\ndebug3: SSH2_FXP_STATUS 0\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 localhost '/bin/sh -c '"'"'chmod u+x /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/ /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py && sleep 0'"'"''
<localhost> (0, '', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')

<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 -tt localhost '/bin/sh -c '"'"'/usr/bin/python /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/vmware_guest_find.py && sleep 0'"'"''

<localhost> (1, '\r\n{"msg": "PyVmomi Python module required. Install using \\"pip install PyVmomi\\"", "failed": true, "invocation": {"module_args": {"username": "xxx.xxx@xxx.com", "datacenter": null, "name": "xxxxxxxvm2", "hostname": "xxxxx.xxxxx.net", "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", "validate_certs": false, "port": 443, "uuid": null}}}\r\n', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 1\r\nShared connection to localhost closed.\r\n')
<localhost> ESTABLISH SSH CONNECTION FOR USER: root
<localhost> SSH: EXEC ssh -vvv -o ControlMaster=auto -o ControlPersist=30m -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=120 -o ControlPath=/root/.ansible/cp/e07565dfb4 localhost '/bin/sh -c '"'"'rm -f -r /root/.ansible/tmp/ansible-tmp-1534398031.45-118622861370583/ > /dev/null 2>&1 && sleep 0'"'"''
<localhost> (0, '', 'OpenSSH_7.4p1, OpenSSL 1.0.2k-fips  26 Jan 2017\r\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: /etc/ssh/ssh_config line 58: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2484\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\n')
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "datacenter": null,
            "hostname": "xxxxx.xxxxx.net",
            "name": "xxxxxxxvm2",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 443,
            "username": "xxx.xxx@xxx.com",
            "uuid": null,
            "validate_certs": false
        }
    },
    "msg": "PyVmomi Python module required. Install using \"pip install PyVmomi\""
}
        to retry, use: --limit @/etc/ansible/Vm_rollback_snapshost.retry

PLAY RECAP **********************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1
```

**Comment by Akasurde** [comment]
@ChrisD1990 Thanks for reporting this issue. I see you are installing `PyVmomi` is installed in `/usr/lib64/python2.7/site-packages` whereas Ansible module is looking library in `/usr/lib/python2.7/site-packages`

Can you check this on your setup and let me know ? Thanks,

needs_info

**Comment by ChrisD1990** [comment]
i have two ansible env, i don't know why other env can run PyVmomi, and i check the PyVmomi install path is the same and ansible with /usr/lib/python2.7/site-packages.
```
> [root@Georgeoelv1 ~]# pip show PyVmomi
> Name: pyvmomi
> Version: 6.7.0
> Summary: VMware vSphere Python SDK
> Home-page: https://github.com/vmware/pyvmomi
> Author: VMware, Inc.
> Author-email: jhu@vmware.com
> License: License :: OSI Approved :: Apache Software License
> Location: /usr/lib64/python2.7/site-packages
> Requires: six, requests
> Required-by:
```
other ansible env:
```
[root@ansible001 ~]# python -m site
sys.path = [
    '/root',
    '/usr/lib64/python27.zip',
    '/usr/lib64/python2.7',
    '/usr/lib64/python2.7/plat-linux2',
    '/usr/lib64/python2.7/lib-tk',
    '/usr/lib64/python2.7/lib-old',
    '/usr/lib64/python2.7/lib-dynload',
    '/usr/lib64/python2.7/site-packages',
    '/usr/lib64/python2.7/site-packages/gtk-2.0',
    '/usr/lib/python2.7/site-packages',
    '/usr/lib/python2.7/site-packages/pysftp-0.2.9-py2.7.egg',
    '/usr/lib/python2.7/site-packages/paramiko-2.0.2-py2.7.egg',
]
USER_BASE: '/root/.local' (exists)
USER_SITE: '/root/.local/lib/python2.7/site-packages' (doesn't exist)
ENABLE_USER_SITE: True

[root@ansible001 ~]# ansible --version
ansible 2.5.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
[root@ansible001 ~]# pip show PyVmomi
Name: pyvmomi
Version: 6.7.0
Summary: VMware vSphere Python SDK
Home-page: https://github.com/vmware/pyvmomi
Author: VMware, Inc.
Author-email: jhu@vmware.com
License: License :: OSI Approved :: Apache Software License
Location: /usr/lib64/python2.7/site-packages
Requires: requests, six
Required-by:
```

**Comment by Im0** [comment]
Hi @ChrisD1990,

As @Akasurde points out, it looks like there is a discrepancy with the path.  pyvmomi appears to be installed under `/usr/lib64/python2.7` yet Ansible is looking in `/usr/lib/python2.7/site-packages` for pyvmomi.

I was thinking, perhaps try:

1. Run `PYTHONPATH=/usr/lib64/python2.7 ansible-playbook /etc/ansible/Vm_rollback_snapshost.yml`

Does that work?

2. Otherwise, perhaps create a python virtual environment for testing:

File: create_venv.yml
```
---
- name: Create venv
  hosts: localhost
  tasks:
    - pip:
        name: pyvmomi,ansible
        virtualenv: /tmp/pyvmomi_venv
        virtualenv_site_packages: no
        state: latest
```
Run:
* ansible-playbook create_venv.yml
* `. /tmp/pyvmomi_venv/bin/activate`  - Activates the python virtual environment 
* `pip list`  - Should show ansible and pyvmomi installed
* `ansible --version` - Should show version 2.6.x and:
```
  ansible python module location = /tmp/pyvmomi_venv/local/lib/python2.7/site-packages/ansible
  executable location = /tmp/pyvmomi_venv/bin/ansible

```

Then, try running your playbook again... see what happens.  

When done, exit out of the python virtual environment with `deactivate`.

**Comment by ChrisD1990** [comment]
Sorry for not replying in time, i will try your advice. thanks all..

**Comment by ChrisD1990** [comment]
@Im0 and @Akasurde 
i try these, but still can not success. i don't know why always prompt me need install PyVmomi. So i revert my ansible vm , but still failed. can you tell me ansible how to find the pyvmomi model? 

i use host is localhost or my vm ip on yml file,  Is it related to the host I specified?

**---
  - name: Revert the given snapshot for virtual machines
    hosts: localhost
    tasks:
      - name: get vm state
        vmware_guest_find:
          hostname: "{{ vcenter_hostname }}"
          username: "{{ vsphere_user }}"
          password: "{{ vsphere_pwd }}"
          name: "{{ vm_name }}"
          validate_certs: False
        register: vm_state
        delegate_to: localhost
#        failed_when: vm_state is failure

      - name: Display all variables/facts known for a host
        debug:
          var: vm_state**


[root@Georgeoelv1 ansible]# pip install PyVmomi
Requirement already satisfied: PyVmomi in /usr/lib/python2.7/site-packages (6.7.0)
Requirement already satisfied: six>=1.7.3 in /usr/lib/python2.7/site-packages (from PyVmomi) (1.9.0)
Requirement already satisfied: requests>=2.3.0 in /usr/lib/python2.7/site-packages (from PyVmomi) (2.19.1)
Requirement already satisfied: idna<2.8,>=2.5 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (2.7)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (3.0.4)
Requirement already satisfied: urllib3<1.24,>=1.21.1 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (1.23)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (2018.4.16)
rtslib-fb 2.1.63 has requirement pyudev>=0.16.1, but you'll have pyudev 0.15 which is incompatible.
ipapython 4.5.4 has requirement dnspython>=1.15, but you'll have dnspython 1.12.0 which is incompatible.
You are using pip version 10.0.1, however version 18.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[root@Georgeoelv1 ansible]# pip show PyVmomi
Name: pyvmomi
Version: 6.7.0
Summary: VMware vSphere Python SDK
Home-page: https://github.com/vmware/pyvmomi
Author: VMware, Inc.
Author-email: jhu@vmware.com
License: License :: OSI Approved :: Apache Software License
Location: /usr/lib/python2.7/site-packages
Requires: six, requests
Required-by:
You are using pip version 10.0.1, however version 18.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
[root@Georgeoelv1 ansible]# pip install --upgrade pip
Collecting pip
  Downloading https://files.pythonhosted.org/packages/5f/25/e52d3f31441505a5f3af41213346e5b6c221c9e086a166f3703d2ddaf940/pip-18.0-py2.py3-none-any.whl (1.3MB)
    100% |████████████████████████████████| 1.3MB 582kB/s
rtslib-fb 2.1.63 has requirement pyudev>=0.16.1, but you'll have pyudev 0.15 which is incompatible.
ipapython 4.5.4 has requirement dnspython>=1.15, but you'll have dnspython 1.12.0 which is incompatible.
Installing collected packages: pip
  Found existing installation: pip 10.0.1
    Uninstalling pip-10.0.1:
      Successfully uninstalled pip-10.0.1
Successfully installed pip-18.0
[root@Georgeoelv1 ansible]# pip install PyVmomi
Requirement already satisfied: PyVmomi in /usr/lib/python2.7/site-packages (6.7.0)
Requirement already satisfied: six>=1.7.3 in /usr/lib/python2.7/site-packages (from PyVmomi) (1.9.0)
Requirement already satisfied: requests>=2.3.0 in /usr/lib/python2.7/site-packages (from PyVmomi) (2.19.1)
Requirement already satisfied: idna<2.8,>=2.5 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (2.7)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (3.0.4)
Requirement already satisfied: urllib3<1.24,>=1.21.1 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (1.23)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python2.7/site-packages (from requests>=2.3.0->PyVmomi) (2018.4.16)
[root@Georgeoelv1 ansible]# ansible-playbook Vm_rollback_snapshost.yml  --extra-vars "vcenter_hostname=selvc01.hpeswlab.net vsphere_user=asiapacific\dongch vsphere_pwd='Admin@1990lsls' vm_name=shcNEWcos74v1 vm_snapshot_name=old_shcnewcos74v1"

PLAY [Revert the given snapshot for virtual machines] ***************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************************************************
ok: [16.186.77.61]

TASK [get vm state] *************************************************************************************************************************************************************************************************
fatal: [16.186.77.61 -> localhost]: FAILED! => {"changed": false, "msg": "PyVmomi Python module required. Install using \"pip install PyVmomi\""}
        to retry, use: --limit @/etc/ansible/Vm_rollback_snapshost.retry

PLAY RECAP **********************************************************************************************************************************************************************************************************
16.186.77.61               : ok=1    changed=0    unreachable=0    failed=1

**Comment by ChrisD1990** [comment]
@Akasurde and @Im0 
i try these, but still can not success. i don't know why always prompt me need install PyVmomi. So i revert my ansible vm , but still failed. can you tell me ansible how to find the pyvmomi model?

i use host is localhost or my vm ip on yml file, Is it related to the host I specified?

- ---
-   - name: Revert the given snapshot for virtual machines
-     hosts: localhost
-     tasks:
-       - name: get vm state
-         vmware_guest_find:
-           hostname: "{{ vcenter_hostname }}"
-           username: "{{ vsphere_user }}"
-           password: "{{ vsphere_pwd }}"
-           name: "{{ vm_name }}"
-           validate_certs: False
-         register: vm_state
-         delegate_to: localhost
- #        failed_when: vm_state is failure
- 
-       - name: Display all variables/facts known for a host
-         debug:
-           var: vm_state

**pyvmomi version and install path**
[root@Georgeoelv1 ansible]# pip show PyVmomi
Name: pyvmomi
Version: 6.7.0
Summary: VMware vSphere Python SDK
Home-page: https://github.com/vmware/pyvmomi
Author: VMware, Inc.
Author-email: jhu@vmware.com
License: License :: OSI Approved :: Apache Software License
Location: /usr/lib/python2.7/site-packages
Requires: six, requests
Required-by:
You are using pip version 10.0.1, however version 18.0 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

**Ansible version**

[root@Georgeoelv1 ansible]# ansible --version
ansible 2.5.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Feb 20 2018, 09:19:12) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

[root@Georgeoelv1 ansible]# ansible-playbook Vm_rollback_snapshost.yml --extra-vars "vcenter_hostname=xccdd.vm.net vsphere_user=chrisd1990 vsphere_pwd='Admin@aassddsasdsdfs' vm_name=cos74v1 vm_snapshot_name=old_cos74v1"

PLAY [Revert the given snapshot for virtual machines] ***************************************************************************************************************************************************************

TASK [Gathering Facts] **********************************************************************************************************************************************************************************************
ok: [localhost]

TASK [get vm state] *************************************************************************************************************************************************************************************************
fatal: [localhost -> localhost]: FAILED! => {"changed": false, "msg": "PyVmomi Python module required. Install using "pip install PyVmomi""}
to retry, use: --limit @/etc/ansible/Vm_rollback_snapshost.retry

PLAY RECAP **********************************************************************************************************************************************************************************************************
localhost : ok=1 changed=0 unreachable=0 failed=1

**Comment by Im0** [comment]
@ChrisD1990 could you please try running this on the command line: 

```
shell$ python
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from pyVmomi import vim, vmodl
>>> 

```

Ansible loads pyVmomi like `from pyVmomi import vim, vmodl`.  I'd like to see if there are any issues at this level for you.

Also, what OS are you on?

**Comment by ChrisD1990** [comment]
[root@Georgeoelv1 ansible]# python
Python 2.7.5 (default, Feb 20 2018, 09:19:12)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-28)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from pyVmomi import vim, vmodl
>>>
>>> exit()

[root@Georgeoelv1 ansible]# cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.5 (Maipo)

[root@Georgeoelv1 ansible]# uname -a
Linux Georgeoelv1.hpeswlab.net 3.10.0-862.6.3.el7.x86_64 #1 SMP Fri Jun 15 17:57:37 EDT 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by ChrisD1990** [comment]
i can import pyvmomi with python, but i don't know why ansible-playbook can't find the model.

**Comment by ChrisD1990** [comment]
@Im0 and @Akasurde  i re-setup the ansible env, now ansible is worked for pyVmomi. shall i close the issue?

**Comment by Im0** [comment]
Hi @chrisd1990

I think so.  It will be hard to troubleshoot now the problem has gone.

I’m wondering if ansible.cfg was modified.

Anyhow,  close it down, re-open if you encounter the issue again.

Cheers Im0

**Comment by ChrisD1990** [comment]
I closed the issue, thanks all for your help.

**Comment by fedelemantuano** [comment]
I have the same issue. I installed `ansible 2.6.3` and `PyVmomi` in a virtualenv. When I run the playbook, I get the same error.
Then I installed `PyVmomi` in system without virtualenv and It worked.
So I can import in virtualenv the library but I can't use it in virtualenv.

**Comment by Im0** [comment]
@fedelemantuano may I suggest opening a new issue and include debugging output from your environment?  It’s be good to see your setup and environment variables.

Please reference this issue in your notes so we can see the link.

**Comment by fedelemantuano** [comment]
ok @Im0 thanks for reply.

**Comment by jjahns** [comment]
Hi,

I am reopening this for the folks with issues running Ansible in a virtualenv and then trying to use Pyvmomi.

You need to edit your ansible.cfg and set the ansible_python_interpreter value to the absolute path of your python executable INSIDE of your virtualenv.

It defaults to the /usr/bin/python value.

And /usr/bin/python will NOT find your Pyvmomi install if you installed it in a virtualenv.

I can reproduce this exact behavior and can confirm that by changing that value (I'd recommend creating a host_vars or group_vars for your localhost instead) fixes the problem.

**Comment by rbcoach** [comment]
Hi,

i found solution for using virtualenv (for me and my colleagues are working). Add to your playbook this:
```
vars:
  ansible_python_interpreter: /home/rbcoach/.virtualenvs/ansible-2.6.2/bin/python
```
Change to your path where you have virtualenvs installed (full path to python binary). Of course you must install package over pip in virtualenv.

**Comment by karthi81** [comment]
Hi,

I am running into PyVmomi  module loading issue.

Error Message:
```
"msg": "PyVmomi Python module required. Install using \"pip install PyVmomi\""
```
Output of ansible --version: 
==================
```
(dc_infra_v2_lib) [vagrant@st2vagrant ansible]$ ansible --version
ansible 2.7.3
  config file = /opt/stackstorm/packs/dc_infra_v2_lib/dc_infra_v2_lib/ansible/ansible.cfg
  configured module search path = [u'/usr/share/my_modules']
  ansible python module location = /opt/stackstorm/virtualenvs/dc_infra_v2_lib/lib/python2.7/site-packages/ansible
  executable location = /opt/stackstorm/virtualenvs/dc_infra_v2_lib/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
(dc_infra_v2_lib) [vagrant@st2vagrant ansible]$
```
Where I can set the default search path for PyVmomi  module?

Note: I am running simple playbook.

**Comment by Akasurde** [comment]
@karthi81 Are you using Python which is not in default path ?

**Comment by karthi81** [comment]
Thanks for your help here!

Here are a few other details:

I have VM and Ansible installed on different Virtual Env. I am getting the error when I try to use delegate_local parameter to connect with VCenter.

Sample Code:
- name: Gather facts about ESXi Host
  vmware_host_config_facts:
    hostname: '{{ vcenter_hostname }}'
    username: '{{ vcenter_username }}'
    password: '{{ vcenter_password }}'
    esxi_hostname: '{{ esxi_hostname }}'
  delegate_to: localhost

then I did install the ansible local as well via pip (Not in any python virtual Env) and it's working but how can I use the custom virtual env ansilble location like /opt/stackstorm/pakcs/ansible and I did not find any option in ansible.cfg or I may not aware of the exact option.

Let me know if you have any questions.

**Comment by mrlesmithjr** [comment]
Just leaving my findings right here with a scenario I just ran into where `PyVmomi` was installed but Ansible threw an error stating to install it. I spent a good 2 hours going through each and every library install in my virtualenv until I found the culprit. My issue was that `PyVmomi` was indeed installed but so was `PyVim`, once I uninstalled `PyVim` the issue went away. Just thought I would share this here.

**Comment by thanos-k** [comment]
I had the exact same incomprehensible error message 
and the actual problem was that localhost (which we delegate to) was not part of the inventory

As soon as I added localhost to the inventory file the playbook worked like a charm. 
Weird error message though.
